File: lib/aerox/scene/components/PointLightComponent.cpp
#include "aerox/drawing/scene/types.hpp"

#include <aerox/scene/components/PointLightComponent.hpp>
namespace aerox::scene {

drawing::GpuLight PointLightComponent::GetLightInfo() {
  drawing::GpuLight info{};
  info.color = glm::vec4{1.0,1.0,1.0,_intensity};
  const auto worldLocation = GetWorldLocation();
  info.location = glm::vec4{worldLocation.x,worldLocation.y,worldLocation.z,0.0f};
  info.direction = glm::vec4{0.0f,-1.0f,0.0f,0.0};
  return info;
}
}

File: old/aerox.Runtime.Scene/Components/Lights/PointLightComponent.cs
using aerox.Runtime.Math;
using aerox.Runtime.Scene.Graphics;

namespace aerox.Runtime.Scene.Components.Lights;

public class PointLightComponent : LightComponent
{
    public override SceneLight ToSceneLight()
    {
        return new SceneLight()
        {
            Color = new Vector4<float>(Color.X,Color.Y,Color.Z,Intensity),
            Direction = 0.0f,
            Location = new Vector4<float>(GetWorldTransform().Location, 1.0f)
        };
    }
}